packages/packages/core/editor-canvas/src/legacy/create-templated-element-type.ts

```diff
@@ -91,14 +79,24 @@ export function createTemplatedElementView( {
 			this.render();
 		}
 
-		// Overriding Marionette original render method to inject our renderer.
-		async _renderTemplate() {
-			this.#beforeRenderTemplate();
-
+		render() {
 			this.#abortController?.abort();
 			this.#abortController = new AbortController();
 
 			const process = signalizedProcess( this.#abortController.signal )
+				.then( () => this.#beforeRender() )
+				.then( () => this._renderTemplate() )
+				.then( () => this._renderChildren() ) // This will render children and should be async as well.
+				.then( () => this.#afterRender() );
+
+			return process.execute();
+		}
```

Comment:
I know the full explanation is too long for a comment, but maybe just add something like
```
// We override the render function to support async _renderTemplate
// Note that `_renderChildren` asynchronity is still NOT supported, so only the parent element rendering can be async
```

Maybe we should also remove the separate `then` so it will be clear
```suggestion
		render() {
			this.#abortController?.abort();
			this.#abortController = new AbortController();

			const process = signalizedProcess( this.#abortController.signal )
				.then( () => this.#beforeRender() )
				.then( () => this._renderTemplate() )
				.then( () => {
				    this._renderChildren();
				    this.#afterRender();
				 });

			return process.execute();
		}
```